enterprise/tpa_pipeline.py

```diff
@@ -62,7 +63,10 @@ def handle_enterprise_logistration(backend, user, **kwargs):
         return
 
     # proceed with the creation of a link between the user and the enterprise customer, then exit.
-    EnterpriseCustomerUser.objects.update_or_create(
+    __, created = EnterpriseCustomerUser.objects.update_or_create(
         enterprise_customer=enterprise_customer,
         user_id=user.id
     )
+    if created:
+        # Announce the registration
+        REGISTER_ENTERPRISE_USER.send(sender=None, user=user)
```

Comment:
Another approach is to use the post_save signal/receiver instead of using the custom one as doing here.

Reply:
Yes, no need for the custom signal.